### PR TITLE
Add `gluonts.itertools.Chain`

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Add a thin wrapper around `itertools.chain`. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/itertools.py [hunk 1]
```diff
@@ -128,6 +129,23 @@ def get_batch():
...
+    This is a thin wrapper around ``itertools.chain``.
+    """
+
+    iterables: Collection[Iterable]
+
+    def __iter__(self):
+        yield from itertools.chain(*self.iterables)
+
+    def __len__(self) -> int:
+        return sum(len(iterable) for iterable in self.iterables)
```
Comment (jaheba) [inline src/gluonts/itertools.py]:
```suggestion
        return sum(map(len, self.iterables))
```

File: src/gluonts/itertools.py [hunk 2]
```diff
@@ -128,6 +129,23 @@ def get_batch():
     return iter(get_batch, [])
 
 
+@dataclass
+class Chained:
```
Comment (jaheba) [inline src/gluonts/itertools.py]: I think it's nicer to use the same name as in `itertools`.

```suggestion
class Chain:
```

File: test/test_itertools.py [hunk 3]
```diff
@@ -320,3 +321,18 @@ def test_random_yield(
...
+@pytest.mark.parametrize(
+    "iterables",
+    [
+        [[1, 2, 3], [4, 5, 6, 7]],
+    ],
+)
+def test_chained(iterables: List[Iterable]):
+    expected = list(itertools.chain(*iterables))
+    chained = Chained(iterables)
+    _ = list(chained)
```
Comment (jaheba) [inline test/test_itertools.py]: Why not just compare twice?

File: src/gluonts/itertools.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/itertools.py]: @jaheba applied the change, although I had chosen `Chained` for grammatical consistency with `Cached`...

File: test/test_itertools.py (hunk 3 above)
Comment (lostella) [reply test/test_itertools.py]: updated

File: src/gluonts/itertools.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/itertools.py]: Maybe `Cache` would also be better? `Cylic` should probably also be `Cycle`, but we have to start somewhere!

Also, I like `Chain` more than `Chained` regardless :).

**Review by jaheba — APPROVED** [review]
Thanks!

**Pull request merged**